Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -546,13 +546,23 @@ def merge_bottle_spec(old_bottle_spec, new_bottle_hash)
     {
       root_url: new_bottle_hash["root_url"],
       prefix:   new_bottle_hash["prefix"],
-      cellar:   new_bottle_hash["cellar"].to_sym,
+      cellar:   new_bottle_hash["cellar"]&.to_sym,
       rebuild:  new_bottle_hash["rebuild"],
     }.each do |key, new_value|
       old_value = old_bottle_spec.send(key)
       next if key == :rebuild && old_value.zero?
-      next if key == :prefix && old_value == Homebrew::DEFAULT_PREFIX
+      next if key == :root_url && [
+        HOMEBREW_BOTTLE_DEFAULT_MACOS_DOMAIN,
+        HOMEBREW_BOTTLE_DEFAULT_LINUX_DOMAIN,
+      ].any? { |domain| old_value.start_with? domain }
+      next if key == :prefix && [
+        Homebrew::DEFAULT_PREFIX,
+        HOMEBREW_DEFAULT_PREFIX,
```

Comment:
```suggestion
        HOMEBREW_DEFAULT_PREFIX,
```

Don't need both of these.

Reply:
It's required for tests as it is redefined here: https://github.com/Homebrew/brew/blob/2c77a54/Library/Homebrew/test/support/lib/config.rb#L53-L54

Should I instead redefine `HOMEBREW_DEFAULT_PREFIX` as well?

Reply:
Hmm, I wonder if we can get it down to a single variable here then by replacing all occurrences of one with the other?

Reply:
Should we investigate more here or can we merge this PR? I would like to open my PR for the new cellar syntax, but as long as all of this code is a moving target it's not possible for me.

Reply:
I'd like to get it sorted before this PR is merged ideally, sorry.